sequential/utils.py

```diff
@@ -243,3 +248,121 @@ def validate_max_span(max_span: Union[int, None]):
         check_true(isinstance(max_span, int), ValueError("Maximum span should be an integer."))
         check_true(max_span > 1, ValueError("Maximum span should be greater than 1."))
 
+
+def validate_min_frequency(num_rows, min_frequency):
+    """
+    Validate min_frequency
+
+    """
+    # Check min_frequency conditions
+    if isinstance(min_frequency, float):
+        check_true(0.0 < min_frequency,
+                   ValueError("Minimum frequency percentage should be greater than 0.0. "
+                              "Current minimum frequency is {}.".format(min_frequency)))
+        check_true(min_frequency <= 1.0, ValueError("Minimum frequency percentage should be less than 1.0. "
+                                                    "Current minimum frequency is {}.".format(min_frequency)))
+        check_true(min_frequency * num_rows >= 1.0, ValueError("Minimum frequency percentage should set the minimum "
+                                                               "row count to be no less than 1.0. "
+                                                               "Thus the percentage should be no less than "
+                                                               "1/(number of sequences)."))
+    elif isinstance(min_frequency, int):
+        check_true(0 < min_frequency, ValueError("Minimum frequency should be greater than 0.0. "
+                                                 "Current minimum frequency is {}.".format(min_frequency)))
+        check_true(min_frequency <= num_rows, ValueError("Minimum frequency cannot be more than number of sequences. "
+                                                         "Current minimum frequency is {}.".format(min_frequency)))
+    else:
+        raise TypeError("Minimum frequency should be integer (as a row count) or float (as a row percentage).")
+
+
+def validate_min_frequency_with_batch(num_rows, batch_size, min_frequency):
+    """
+    Validate min_frequency when Seq2Pat runs on batches
```

Comment:
can you add a word on what's the expecttation for a valid or invalid min_freq for the batch case?

Reply:
> can you add a word on what's the expecttation for a valid or invalid min_freq for the batch case?

A valid min_freq would be a float that makes `batch_size * min_freq` to generate anrow count as an integer greater than 0. 

User can define `batch_size`, while there would be one batch (the last one when we split sequentially) containing less sequences than `batch_size`. So validation step also checks if `min_freq` works with the last batch.